include/irr/asset/CMTLPipelineMetadata.h

```diff
@@ -0,0 +1,99 @@
+#ifndef __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+#define __IRR_C_MTL_PIPELINE_METADATA_H_INCLUDED__
+
+#include "irr/asset/IPipelineMetadata.h"
+
+namespace irr {
+namespace asset
+{
+
+class CMTLPipelineMetadata final : public IPipelineMetadata
+{
+public:
+    struct SMtl
+    {
+        std::string name;
+
+        //Ka
+        core::vector3df_SIMD ambient = core::vector3df_SIMD(1.f);
+        //Kd
+        core::vector3df_SIMD diffuse = core::vector3df_SIMD(1.f);
+        //Ks
+        core::vector3df_SIMD specular = core::vector3df_SIMD(1.f);
+        //Ke
+        core::vector3df_SIMD emissive = core::vector3df_SIMD(1.f);
+        //Tf
+        core::vector3df_SIMD transmissionFilter = core::vector3df_SIMD(1.f);
+        //Ns, specular exponent in phong model
+        float shininess = 32.f;
+        //d
+        float opacity = 1.f;
+        //illum
+        uint32_t illumModel = 0u;
+        //-bm
+        float bumpFactor = 1.f;
```

Comment:
would be nice if we could roll this into a struct that would be ready to be copied into a `std140` UBO